client/main.lua

```diff
@@ -615,26 +658,41 @@ RegisterNetEvent('qbx_vehicleshop:client:swapVehicle', function(data)
     local dataTargetVehicle = sharedConfig.shops[shopName].showroomVehicles[data.targetVehicle]
     if dataTargetVehicle.vehicle == data.toVehicle then return end
 
-    local closestVehicle = lib.getClosestVehicle(dataTargetVehicle.coords.xyz, 5, false)
-    if not closestVehicle then return end
-
     if not IsModelInCdimage(data.toVehicle) then
         lib.print.error(('Failed to find model for "%s". Vehicle might not be streamed?'):format(data.toVehicle))
         return
     end
 
-    DeleteEntity(closestVehicle)
-    while DoesEntityExist(closestVehicle) do
-        Wait(50)
+    local points = lib.points.getAllPoints()
+    local vehPoint = nil
+    for i = 1, #points do
+        local point = points[i]
+        if point.coords == dataTargetVehicle.coords.xyz then
+            vehPoint = point
+            break
+        end
     end
 
-    local veh = createShowroomVehicle(data.toVehicle, dataTargetVehicle.coords)
-
-    dataTargetVehicle.vehicle = data.toVehicle
+    if not vehPoint then return end
 
-    if config.useTarget then
-        createVehicleTarget(shopName, veh, data.targetVehicle)
+    if vehPoint.veh then
+        if DoesEntityExist(vehPoint.veh) then
+            DeleteEntity(vehPoint.veh)
+        end
+        if config.useTarget then
+            exports.ox_target:removeLocalEntity(vehPoint.veh, 'vehicleshop:showVehicleOptions')
+        else
+            vehPoint.boxZone:remove()
+        end
     end
+    vehPoint:remove()
+
+    createShowroomVehiclePoint({
+        coords = dataTargetVehicle.coords,
+        shopName = shopName,
+        vehiclePos = data.targetVehicle,
+        model = data.toVehicle
+    })
```

Comment:
Is it better to recreate *everything*? Can we not replace the vehicle without touching anything else?

Reply:
Would have to probably maintain a separate table of which spot goes to which model and add a check inside the point for while inside change the model. You still have to do the teardown of the ox_target zone when changing vehicles as it is bound to the entity.

I didn't do any testing to see if it has that big of a performance impact but from a code readability standpoint I feel that this is much cleaner to read compared to the previous iteration.

If you have an idea on a better way to do it I can certainly change it. This was just how I figured it could work.